library/src/scala/compiletime/testing/package.scala

```diff
@@ -2,14 +2,16 @@ package scala.compiletime
 package testing
 
 /** Whether the code type checks in the current context?
+ *
+ *  An inline definition with a call to `typeChecks` should be transparent.
  *
  *  @param code The code to be type checked
  *
  *  @return false if the code has syntax error or type error in the current context, otherwise returns true.
  *
  *  The code should be a sequence of expressions or statements that may appear in a block.
  */
-inline def typeChecks(inline code: String): Boolean =
+transparent inline def typeChecks(inline code: String): Boolean =
   // implemented in package dotty.tools.dotc.typer.Inliner.Intrinsics
   error("Compiler bug: `typeChecks` was not checked by the compiler")
```

Comment:
If we handle the intrinsics in `Inlining` phase, it seems we don't need `transparent`. Same for below.

Reply:
`typeChecks` has an interaction with the typer state the might not be possible after typer.

https://github.com/lampepfl/dotty/blob/8443c16b7d5f965c23418d5b13aaa195116842a1/compiler/src/dotty/tools/dotc/typer/Inliner.scala#L294

Reply:
If we have the correct context that contains the right scope with imports, it seems to be possible.

Reply:
We do not as the only typer we have when inlining in the InlineTyper and that one starts when expanding the inline call.

Reply:
Maybe it is the import info that is not set correctly in `Inlining`